python/ray/worker.py

```diff
@@ -1681,13 +1681,13 @@ def cancel(object_id, force=False):
     worker = ray.worker.global_worker
     worker.check_connected()
 
-    if isinstance(object_id, ray.ObjectID) and object_id.task_id().actor_id(
-    ).hex() not in ray.actors().keys():
-        if len(ray.wait([object_id], timeout=0)[1]) != 1:
-            return
-        return worker.core_worker.kill_task(object_id, force)
-    raise ValueError("ray.cancel() only supported for non-actor object IDs. "
-                     "Got: {}.".format(type(object_id)))
+    if not isinstance(object_id, ray.ObjectID):
+        raise ValueError(
+            "ray.cancel() only supported for non-actor object IDs. "
+            "Got: {}.".format(type(object_id)))
+    if len(ray.wait([object_id], timeout=0)[1]) != 1:
```

Comment:
This is to prevent an error later on. I could move this check to CoreWorker?

Reply:
Hmm it seemed like the logic in the CoreWorker should be able to catch this so I thought we could just remove it completely. Is that not possible?

Reply:
The CoreWorker will treat this like a task assigned to a remote host and reply with `"Task is not locally submitted."` because in this case the task has no corresponding `task_spec`. I could replicate the silent return if the object exists in the object_store in CoreWorker.

Reply:
Hmm yeah I think you could check inside the CoreWorker based on the ref counter state. You can check whether the current process is the owner of an object using something like [this](https://github.com/ray-project/ray/blob/master/src/ray/core_worker/core_worker.cc#L698).

Reply:
Sounds good!